test/roundtrip.ts

```
@@ -109,19 +113,176 @@ describe('roundtrip', () => {
       `, {fortuneCookie: 'live long and prosper'});
     });
   });
+
+  describe('fragments', () => {
+    it('should resolve on union types with inline fragments', () => {
+      storeRoundtrip(gql`
+        query {
+          all_people {
+            name
+            ... on Jedi {
+              side
+            }
+            ... on Droid {
+              model
+            }
+          }
+        }`, {
+        all_people: [
+          {
+            name: 'Luke Skywalker',
+            side: 'bright',
+          },
+          {
+            name: 'R2D2',
+            model: 'astromech',
+          },
+        ],
+      });
+    });
+
+    it('should throw on error on two of the same inline fragment types', () => {
+      assert.throws(() =>
+        storeRoundtrip(gql`
+          query {
+            all_people {
+              name
+              ... on Jedi {
+                side
+              }
+              ... on Jedi {
+                rank
+              }
+            }
+          }`, {
+          all_people: [
+            {
+              name: 'Luke Skywalker',
+              side: 'bright',
+            },
+          ],
+        })
+      );
+    });
+
+    it('should resolve on union types with spread fragments', () => {
+      storeRoundtrip(gql`
+        fragment jediFragment on Jedi {
+          side
+        }
+
+        fragment droidFragment on Droid {
+          model
+        }
+
+        query {
+          all_people {
+            name
+            ...jediFragment
+            ...droidFragment
+          }
+        }`, {
+        all_people: [
+          {
+            name: 'Luke Skywalker',
+            side: 'bright',
+          },
+          {
+            name: 'R2D2',
+            model: 'astromech',
+          },
+        ],
+      });
+    });
+
+    it('should throw on error on two of the same spread fragment types', () => {
```

I don't think this should throw an error - don't people use multiple fragments on one type like all the time?